scripts-available/CDB_ForeignTable.sql

```diff
@@ -139,6 +139,15 @@ $$
 LANGUAGE plpgsql VOLATILE PARALLEL UNSAFE;
 
 
+-- Produce a valid DB name for objects created for the user FDW's
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_User_FDW_Object_Names(fdw_input_name NAME)
+RETURNS NAME AS $$
+  -- Note on input we use %s and on output we use %I, in order to
+  -- avoid double escaping
+  SELECT format('cdb_fdw_%s', fdw_input_name)::NAME;
```

Comment:
Should we enclose this also under `quote_ident` to prevent odd things like spaces and possible injections (although the user already has full access, so them being able to inject code shouldn't be a concern :D).

Reply:
I'm applying the principle of "sanitize on input, escaping on output" (couldn't find a good reference to it on the internets, sorry). I also added an intentional comment about that in the code (BTW, `%I` is equivalent to `quote_ident`).

Let me explain it a little further:
- The function takes a `NAME` type, which is meant to host identifiers, but perhaps requires quoting such as in this `valid-name-with-dashes` which has been always a nuance. It could be tempting to use more specific types (and it is always a good idea, when possible), but cannot in this case as those objects do not exist yet. E.g:
```
postgres=# SELECT 'hola'::regnamespace;
ERROR:  schema "hola" does not exist
```

- Wherever I use the generated name, I use `format('%I')` to "escape on output".

- I modified the tests to use dashes in the base name, just to check there's no missing escaping or double escaping.

For the sake of the experiment, let's follow it with a malicious input:

```sql
-- Ok, evil input
SELECT cartodb.__CDB_User_FDW_Object_Names('evil_sufix; DROP TABLE users');
     __cdb_user_fdw_object_names      
--------------------------------------
 cdb_fdw_evil_sufix; DROP TABLE users
(1 row)

-- But escaping "on output"
SELECT format('CREATE SERVER %I FOREIGN DATA WRAPPER postgres_fdw', cartodb.__CDB_User_FDW_Object_Names('evil_sufix; DROP TABLE users'));
                                         format                                         
----------------------------------------------------------------------------------------
 CREATE SERVER "cdb_fdw_evil_sufix; DROP TABLE users" FOREIGN DATA WRAPPER postgres_fdw
(1 row)

-- As horrible as it is, it generated a valid identifier that requires quoting
-- We can run the query...
CREATE SERVER "cdb_fdw_evil_sufix; DROP TABLE users" FOREIGN DATA WRAPPER postgres_fdw;
CREATE SERVER

-- and finally delete it
DROP SERVER "cdb_fdw_evil_sufix; DROP TABLE users";
DROP SERVER
```

Mind that this will require superuser privileges to execute, or it will fail at the first step and the whole transaction (function call or enclosing transaction) will be rolled back.

If you still deem it necessary, the only thing I can think of is to get stricter (and custom) sanitization on input. If you could think of other approaches (not just for this piece of code but more general to the extensions) I'm all ears.

Reply:
> BTW, %I is equivalent to quote_ident

Nice, I didn't know. I'd say that's enough, as going custom for validation is bound to give us a lot of headaches.